### DISCUSSION TITLE
How to add ControlNet?

### DISCUSSION THREAD

**Discussion by CarbonMatrix** [TheLastBen/fast-stable-diffusion#1601] [Q&A]
Anyone?

**Comment by TheLastBen** [comment]
https://www.youtube.com/watch?v=OxFcIv8Gq8o

**Reply by CarbonMatrix** [reply]
Thanks, but need it in google colab and not in local installation.
There have been requests on reddit to implemet the google colab feature in the fast LastBen version.
There is working file from camenduru, maybe you can implement something similar into your ipynb file?
https://colab.research.google.com/github/camenduru/stable-diffusion-webui-colab/blob/v2.0/control_net_openpose_webui_colab.ipynb

**Comment by animeking527** [comment]
Run the webui colab and just follow what is in the video to install the extension & get the models. Only difference is when he refers to the hard drive locations in the video you would do it in your sd folder in your gdrive instead.

**Reply by meunumerotim333** [reply]
How to run webui colab? Didn't find it.

**Reply by animeking527** [reply]
Run the TheLastBen's colab.

https://github.com/TheLastBen/fast-stable-diffusion
Automatic1111

**Reply by meunumerotim333** [reply]
I did, but I still couldn't able to find the webui colab. I'm a noob with codes, just a regular user.

**Reply by animeking527** [reply]
The webui is the interface that you use to interact and make images. Have you been able to generate any images using TheLastBen's colab? If you have that's the webui. If not, search youtube for stable diffusion colab tutorials. It's a bit difficult to explain if you've never run a colab before. 

If you've got the interface running to generate images, just follow the video TheLastBen linked but you don't need to install/update it since the colab did that. Just follow the instructions for the extension tab, download the models, and places them in your sd folder that's in your gdrive.

**Reply by meunumerotim333** [reply]
Ohhh sorry. I thought the webui was a window like a CMD window to type codes. I'll check, thank you.

**Reply by meunumerotim333** [reply]
It worked. Thank you :)

**Comment by meunumerotim333** [comment]
I would also love to have ControlNET on TheLastBen version natively.. The camenduru version is painful to run another models.

**Comment by TheLastBen** [comment]
I'll work on it

**Reply by meunumerotim333** [reply]
Thank you :)) will improve a lot the colab.
